dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIChatCompletionService.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.ML.OnnxRuntimeGenAI;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Services;
+
+namespace Microsoft.SemanticKernel.Connectors.Onnx;
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
+
+/// <summary>
+/// Represents a chat completion service using OnnxRuntimeGenAI.
+/// </summary>
+public sealed class OnnxRuntimeGenAIChatCompletionService : IChatCompletionService, IDisposable
+{
+    private readonly Model _model;
+    private readonly Tokenizer _tokenizer;
+
+    private Dictionary<string, object?> AttributesInternal { get; } = new();
```

Comment:
Are we able to infer model name from the supplied info? If not, should we have the first arg to the extension method be a model id as happens in the other connectors? That can then be put into the attributes.